cql2pgjson/src/main/java/org/z3950/zing/cql/cql2pgjson/CQL2PgJSON.java

```diff
@@ -472,30 +472,54 @@ private SqlSelect toSql(CQLSortNode node) throws QueryValidationException {
       if (modifiers.cqlSort == CqlSort.DESCENDING) {
         desc = " DESC";
       }  // ASC not needed, it's Postgres' default
-
       IndexTextAndJsonValues vals = getIndexTextAndJsonValues(modifierSet.getBase());
-
+      //prepare sort by clause as if the sort by is not the primary key
+      String sortbyClauseText  = wrapInLowerUnaccent(vals.indexText);
+      String sortByClauseIndex = vals.indexJson;
+      //determine if sort is primary key 
+      if(dbTable != null) {
```

Comment:
This null check should go into the getPkColumnName() method.

Reply:
moved this line to the function and returned null if found to be null

Reply:
Thre default pkColumnName is `id`, therefore you should return `id` if dbTable is null.